### ISSUE TITLE
master fails because not always there are plusses in flags:

### ISSUE THREAD

**Issue by phadej** [haskell/cabal#6853]
cabal.project

```
constraints: simple-sendfile +allow-bsd
```

error

```
Invalid flag assignment: allow-bsd
CallStack (from HasCallStack):
  error, called at ./Distribution/ReadE.hs:42:24 in Cabal-3.3.0.0-BuS0aoNOUxNGwNITn6L7bR:Distribution.ReadE
```

**Comment by phadej** [comment]
Fixed in https://github.com/haskell/cabal/pull/6854

**Comment by phadej** [comment]
Resolved by #6853 (fix) and #6858 (regression test).
